### PR TITLE
Fixed #26021 -- Standardize on hanging indent in documentation

### PR DESCRIPTION
**Pull request by kutenai**
Apply Django code style to example code in documents.

### CODE REVIEW

File: docs/intro/tutorial04.txt [hunk 1]
```diff
@@ -81,10 +81,11 @@ create a real version. Add the following to ``polls/views.py``:
             selected_choice = question.choice_set.get(pk=request.POST['choice'])
         except (KeyError, Choice.DoesNotExist):
             # Redisplay the question voting form.
-            return render(request, 'polls/detail.html', {
```
Comment (timgraham) [inline docs/intro/tutorial04.txt]: This is an okay style -- it doesn't result in any flake8 warnings.

File: docs/intro/tutorial05.txt [hunk 2]
```diff
@@ -495,8 +497,11 @@ class:
             """
             create_question(question_text="Future question.", days=30)
             response = self.client.get(reverse('polls:index'))
-            self.assertContains(response, "No polls are available.",
```
Comment (timgraham) [inline docs/intro/tutorial05.txt]: I'd use a single line here

File: docs/ref/contrib/admin/index.txt [hunk 3]
```diff
@@ -615,10 +615,12 @@ subclass::
               color_code = models.CharField(max_length=6)
 
               def colored_name(self):
-                  return format_html('<span style="color: #{};">{} {}</span>',
-                                     self.color_code,
-                                     self.first_name,
-                                     self.last_name)
+                  return format_html(
+                    '<span style="color: #{};">{} {}</span>',
```
Comment (timgraham) [inline docs/ref/contrib/admin/index.txt]: indentation is off here

File: docs/ref/contrib/admin/index.txt [hunk 4]
```diff
@@ -906,13 +910,17 @@ subclass::
...
+                          return super(
+                              AuthDecadeBornListFilter,
+                              self,
+                          ).lookups(request, model_admin)
 
                   def queryset(self, request, queryset):
                       if request.user.is_superuser:
-                          return super(AuthDecadeBornListFilter,
-                              self).queryset(request, queryset)
+                          return super(
```
Comment (timgraham) [inline docs/ref/contrib/admin/index.txt]: I'd use a single line

File: docs/ref/models/expressions.txt [hunk 5]
```diff
@@ -371,7 +372,8 @@ SQL that is generated. Here's a brief example::
                 expression,
                 distinct='DISTINCT ' if distinct else '',
                 output_field=IntegerField(),
-                **extra)
+                **extra,
```
Comment (timgraham) [inline docs/ref/models/expressions.txt]: I think a comma here is actually a syntax error.

File: docs/intro/tutorial05.txt [hunk 6]
```diff
@@ -485,7 +487,7 @@ class:
             response = self.client.get(reverse('polls:index'))
             self.assertQuerysetEqual(
                 response.context['latest_question_list'],
-                ['<Question: Past question.>']
+                ['<Question: Past question.>'],
```
Comment (timgraham) [inline docs/intro/tutorial05.txt]: I wouldn't worry about a trailing comma for asserts

File: docs/howto/static-files/deployment.txt [hunk 7]
```diff
@@ -106,7 +106,7 @@ Here's how this might look in a fabfile::
         project.rsync_project(
             remote_dir = env.remote_static_root,
```
Comment (timgraham) [inline docs/howto/static-files/deployment.txt]: You could remove the spaces are the equal signs while you're here as that's a flake8 violation.

File: docs/ref/contrib/admin/index.txt [hunk 8]
```diff
@@ -906,13 +910,17 @@ subclass::
 
                   def lookups(self, request, model_admin):
                       if request.user.is_superuser:
-                          return super(AuthDecadeBornListFilter,
-                              self).lookups(request, model_admin)
+                          return super(
```
Comment (timgraham) [inline docs/ref/contrib/admin/index.txt]: This looks awkward. Maybe just leave it as is and when we drop Python 2 compatibility and switch to `super().` we can improve.

Comment (carljm) [inline docs/ref/contrib/admin/index.txt]: I talked to @kutenai here at the sprints and recommended this formatting; it doesn't bother me, and I think it's an improvement over the existing formatting. But if you (Tim) feel it's inconsistent with Django practice, I'll defer.

Comment (MarkusH) [inline docs/ref/contrib/admin/index.txt]: Doesn't that even fit on one line?

File: docs/ref/contrib/gis/db-api.txt [hunk 9]
```diff
@@ -120,7 +120,7 @@ raster models::
 
     >>> from django.contrib.gis.gdal import GDALRaster
     >>> rast = GDALRaster({'width': 10, 'height': 10, 'name': 'Canyon', 'srid': 4326,
-    ...                    'scale': [0.1, -0.1]'bands': [{"data": range(100)}]}
+    ...                    'scale': [0.1, -0.1]'bands': [{"data": range(100)}]})
```
Comment (MarkusH) [inline docs/ref/contrib/gis/db-api.txt]: Well spotted!

File: docs/ref/contrib/gis/tutorial.txt [hunk 10]
```diff
@@ -452,11 +452,21 @@ with the following code::
         'mpoly' : 'MULTIPOLYGON',
     }
 
-    world_shp = os.path.abspath(os.path.join(os.path.dirname(__file__), 'data', 'TM_WORLD_BORDERS-0.3.shp'))
+    world_shp = os.path.abspath(
+        os.path.join(
+            os.path.dirname(__file__),
+            'data',
+            'TM_WORLD_BORDERS-0.3.shp'),
```
Comment (MarkusH) [inline docs/ref/contrib/gis/tutorial.txt]: The trailing `)` here is wrong. But bottom line, I don't mind if we either stick to the current implementation or would go with
```python
world_shp = os.path.abspath(
    os.path.join(a, b, c)
)
```

File: docs/ref/contrib/messages.txt [hunk 11]
```diff
@@ -320,8 +320,12 @@ Adding extra message tags
 For more direct control over message tags, you can optionally provide a string
 containing extra tags to any of the add methods::
 
-    messages.add_message(request, messages.INFO, 'Over 9000!',
-                         extra_tags='dragonball')
+    messages.add_message(
```
Comment (MarkusH) [inline docs/ref/contrib/messages.txt]: This certainly fits in one line.

File: docs/ref/contrib/sites.txt [hunk 12]
```diff
@@ -197,10 +197,14 @@ Here's an example of what the form-handling view looks like::
...
-        send_mail('Thanks for subscribing to %s alerts' % current_site.name,
-            'Thanks for your subscription. We appreciate it.\n\n-The %s team.' % current_site.name,
+        send_mail(
+            'Thanks for subscribing to %s alerts' % current_site.name,
+            'Thanks for your subscription. We appreciate it.\n\n-The %s team.' % (
+                current_site.name,
+            ),
             'editor@%s' % current_site.domain,
-            [user.email])
+            [user.email]
```
Comment (MarkusH) [inline docs/ref/contrib/sites.txt]: trailing `,`?

File: docs/ref/contrib/admin/index.txt (hunk 4 above)
Comment (kutenai) [inline docs/ref/contrib/admin/index.txt]: A single line, with the .lookups is 100 chars in the docs.
The original did NOT have this on a single line, I just applied the style (as I understand it) to the code.

File: docs/ref/models/expressions.txt (hunk 5 above)
Comment (kutenai) [inline docs/ref/models/expressions.txt]: I put together a small example to try and test this.. and it did work.
What leads you to believe the trailing comma is a syntax error? The **extra?

```python
class BaseDummy(object):
    def __init__(self, posarg, **args):
        for k,v in args.items():
            print("K:{} Value:{}".format(k, v))
            
class Count(BaseDummy):

    def __init__(self, expression, distinct=False, **extra):
        super(Count, self).__init__(
            expression,
            distinct='DISTINCT ' if distinct else '',
            output_field="",
            **extra,
        )

if __name__ == "__main__":
    c = Count(r'.*', distinct=True, a=1, b=2, c=3)



```

File: docs/howto/static-files/deployment.txt (hunk 7 above)
Comment (kutenai) [inline docs/howto/static-files/deployment.txt]: Resolved

File: docs/ref/contrib/admin/index.txt (hunk 8 above)
Comment (kutenai) [inline docs/ref/contrib/admin/index.txt]: Move to a single line

File: docs/ref/contrib/gis/tutorial.txt (hunk 10 above)
Comment (kutenai) [inline docs/ref/contrib/gis/tutorial.txt]: Fits on a single line using MarkusH's recommendation

File: docs/ref/contrib/messages.txt (hunk 11 above)
Comment (kutenai) [inline docs/ref/contrib/messages.txt]: Converted to singe line

File: docs/ref/models/expressions.txt (hunk 5 above)
Comment (tobiasmcnulty) [inline docs/ref/models/expressions.txt]: That might not be a syntax error but it does suggest that one could add additional items after `**extra`, which would be a syntax error.

File: docs/howto/static-files/deployment.txt [hunk 13]
```diff
@@ -104,9 +104,9 @@ Here's how this might look in a fabfile::
     def deploy_static():
         local('./manage.py collectstatic')
         project.rsync_project(
-            remote_dir = env.remote_static_root,
-            local_dir = env.local_static_root,
-            delete = True
+            remote_dir=env.remote_static_root,
+            local_dir=env.local_static_root,
+            delete = True,
```
Comment (tobiasmcnulty) [inline docs/howto/static-files/deployment.txt]: remove spaces here too

File: docs/intro/tutorial04.txt (hunk 1 above)
Comment (tobiasmcnulty) [inline docs/intro/tutorial04.txt]: agreed, the original style is preferred IMHO

**Comment by kutenai** [comment]
Rebased and squashed onto master.

File: docs/ref/contrib/gis/db-api.txt (hunk 9 above)
Comment (tobiasmcnulty) [inline docs/ref/contrib/gis/db-api.txt]: still missing a comma before `'bands'`

File: docs/ref/contrib/gis/db-api.txt [hunk 14]
```diff
@@ -129,7 +129,7 @@ Note that this equivalent to::
     >>> dem = Elevation.objects.create(
     ...     name='Canyon',
     ...     rast={'width': 10, 'height': 10, 'name': 'Canyon', 'srid': 4326,
-    ...           'scale': [0.1, -0.1]'bands': [{"data": range(100)}]}
+    ...           'scale': [0.1, -0.1]'bands': [{"data": range(100)}]},
```
Comment (tobiasmcnulty) [inline docs/ref/contrib/gis/db-api.txt]: missing a comma before `bands`

File: docs/releases/1.8.txt [hunk 15]
```diff
@@ -1343,8 +1343,12 @@ the ``url`` that references :func:`django.contrib.sitemaps.views.sitemap`::
...
     from django.contrib.sitemaps.views import sitemap
 
-    url(r'^sitemap\.xml$', sitemap, {'sitemaps': sitemaps},
-        name='django.contrib.sitemaps.views.sitemap')
+    url(
+        r'^sitemap\.xml$',
+        sitemap,
+        {'sitemaps': sitemaps},
+        name='django.contrib.sitemaps.views.sitemap',
+    )
```
Comment (tobiasmcnulty) [inline docs/releases/1.8.txt]: I have mixed feelings about this... advocating this lengthy format for urls.py will result in very long URLs files (most often I see urls formatted like the original code).

File: docs/topics/forms/formsets.txt [hunk 16]
```diff
@@ -673,10 +673,14 @@ a look at how this might be accomplished::
         else:
             article_formset = ArticleFormSet(prefix='articles')
             book_formset = BookFormSet(prefix='books')
-        return render(request, 'manage_articles.html', {
-            'article_formset': article_formset,
-            'book_formset': book_formset,
-        })
+        return render(
```
Comment (tobiasmcnulty) [inline docs/topics/forms/formsets.txt]: The original format is preferred here.

File: docs/releases/1.8.txt (hunk 15 above)
Comment (timgraham) [inline docs/releases/1.8.txt]: I'm fine keeping the original format. I wouldn't worry about reformatting any release notes anyway.

**Comment by timgraham** [comment]
merged in 4a4d7f980e2a66756e1e424f7648dcd28ff765b7, thanks!

I modified a few places that put each arg on its own line -- I think it's okay to put multiple args on the same line in some cases to keep things a bit shorter.

**Pull request closed without merging**